docs/features/writing-different-telemetry-types.md

```diff
@@ -72,14 +72,14 @@ var telemetryContext = new Dictionary<string, object>
 };
 
 // Start measuring
-var startTime = DateTimeOffset.UtcNow;
-durationMeasurement.Start();
-
-// Interact with database
-var products = await _repository.GetProducts();
-
-_logger.LogSqlDependency("sample-server", "sample-database", "my-table", "get-products", isSuccessful: true, startTime: startTime, duration: durationMeasurement.Elapsed, context: telemetryContext);
-// Output: "SQL Dependency sample-server for sample-database/my-table for operation get-products in 00:00:01.2396312 at 03/23/2020 09:32:02 +00:00 (Successful: True - Context: [Catalog, Products], [Tenant, Contoso])"
+using (var measurement = DependencyMeasurement.Start("get-products"))
+{
+    // Interact with database
+    var products = await _repository.GetProducts();
+    
+    _logger.LogSqlDependency("sample-server", "sample-database", "my-table", "get-products", isSuccessful: true, startTime: measurement.StartTime, duration: measurement.Elapsed, context: telemetryContext);
```

Comment:
We need to provide an overload as discussed which also allows you to do this:
`_logger.LogSqlDependency("sample-server", "sample-database", "my-table", "get-products", true, measurement, context: telemetryContext);`

Reply:
Aha.